controllers/disruption_schedule_controller.go

```diff
@@ -0,0 +1,201 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023 Datadog, Inc.
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	chaosv1beta1 "github.com/DataDog/chaos-controller/api/v1beta1"
+	"go.uber.org/zap"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type DisruptionScheduleReconciler struct {
+	Client  client.Client
+	Scheme  *runtime.Scheme
+	BaseLog *zap.SugaredLogger
+	log     *zap.SugaredLogger
+}
+
+const (
+	ScheduledAtAnnotation       = chaosv1beta1.GroupName + "/scheduled-at"
+	DisruptionScheduleNameLabel = chaosv1beta1.GroupName + "/disruption-schedule-name"
+)
+
+func (r *DisruptionScheduleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (res ctrl.Result, err error) {
+	r.log = r.BaseLog.With("scheduleNamespace", req.Namespace, "scheduleName", req.Name)
+	r.log.Info("Reconciling DisruptionSchedule")
+
+	instance := &chaosv1beta1.DisruptionSchedule{}
+
+	// retrieve schedule
+	if err := r.Client.Get(context.Background(), req.NamespacedName, instance); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	if !instance.DeletionTimestamp.IsZero() {
+		// NOTE: add a finalizer if anything needs to be written here
```

Comment:
It can be a TODO for later, but I do think we'll want a finalizer. If anyone deletes the schedule, we will want to delete any ongoing disruptions. That... should be somewhat easy to do, as they'll be in the same namespace, so we can make the disruption a child of this resource, and I think the Delete will just propagate down?